apps/cowswap-frontend/src/utils/getCurrentChainIdFromUrl.ts

```diff
@@ -1,13 +1,20 @@
 import { SupportedChainId } from '@cowprotocol/cow-sdk'
 
+const chainNameToIdMap: { [key: string]: SupportedChainId } = {
+  mainnet: SupportedChainId.MAINNET,
+  gnosis_chain: SupportedChainId.GNOSIS_CHAIN,
+  goerli: SupportedChainId.GOERLI,
+}
+
 export function getCurrentChainIdFromUrl(): SupportedChainId {
   // Trying to get chainId from URL (#/100/swap)
   // eslint-disable-next-line no-restricted-globals
   const { location } = window
   const urlChainIdMatch = location.hash.match(/^#\/(\d{1,9})\D/)
-  const chainId = +(
-    (urlChainIdMatch ? urlChainIdMatch[1] : new URLSearchParams(location.hash.split('?')[1]).get('chain')) || ''
-  )
+  const searchParams = new URLSearchParams(location.hash.split('?')[1])
+  const chainQueryParam = searchParams.get('chain')
+
+  const chainId = +(urlChainIdMatch?.[1] || chainNameToIdMap[chainQueryParam || ''] || '')
```

Comment:
I wrote `chainNameToIdMap[chainQueryParam || '']` instead of `chainNameToIdMap[chainQueryParam]` just to satisfy typescript

<img width="400" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/076b5a58-abb5-4ce0-9ab1-0f8a1cdffa78">